src/gui/programupdater.cpp

```diff
@@ -73,10 +66,17 @@ void ProgramUpdater::checkForUpdates()
 void ProgramUpdater::rssDownloadFinished(const QString &url, const QByteArray &data)
 {
     Q_UNUSED(url);
-
     qDebug("Finished downloading the new qBittorrent updates RSS");
-    QString version;
 
+#ifdef Q_OS_MAC
+    const QString OS_TYPE {"Mac OS X"};
+#elif defined(Q_OS_WIN)
+    const QString OS_TYPE {((QSysInfo::windowsVersion() >= QSysInfo::WV_WINDOWS7)
+            && QSysInfo::currentCpuArchitecture().endsWith("64"))
```

Comment:
[Qt docs](https://doc.qt.io/qt-5/qsysinfo.html#currentCpuArchitecture):
>Returns the architecture of the CPU that the application is running on, in text format. Note that this function depends on what the OS will report and may not detect the actual CPU architecture if the OS hides that information or is unable to provide it. **For example, a 32-bit OS running on a 64-bit CPU is usually unable to determine the CPU is actually capable of running 64-bit programs.**


Could the bold part be true for Windows? Have you tested it?

Reply:
Damn it. Forget the above comment. You're coding for when we run in 64bit OS.